src/renderers/shared/fiber/__tests__/ReactIncrementalRoot-test.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+var React;
+var ReactNoop;
+
+describe('ReactIncrementalRoot', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    ReactNoop = require('react-noop-renderer');
+  });
+
+  function span(prop) {
+    return {type: 'span', children: [], prop};
+  }
+
+  it('prerenders roots', () => {
+    const root = ReactNoop.create();
+    const work = root.prerender(<span prop="A" />);
+    expect(root.getChildren()).toEqual([]);
+    work.commit();
+    expect(root.getChildren()).toEqual([span('A')]);
+  });
+
+  it('resolves `then` callback synchronously if tree is already completed', () => {
```

Comment:
Hmm, is this behavior really necessary for async updates? Or is it only relevant to sync?